ArduCopter/mode.cpp

```diff
@@ -420,6 +420,7 @@ bool Copter::set_mode(const uint8_t new_mode, const ModeReason reason)
 void Copter::update_flight_mode()
 {
     surface_tracking.invalidate_for_logging();  // invalidate surface tracking alt, flight mode will set to true if used
+    attitude_control->landed_gain_reduction(copter.ap.land_complete); // Adjust gains when landed to attenuate ground oscillation
```

Comment:
I have called this feature here to capture all flight modes in a single line. It seems like a good fit but I would like to get others feedback.

Reply:
is land_complete all that you need?

Reply:
Yes, this is a reasonably conservative test.